kernel/src/client/arrow_data.rs

```diff
@@ -170,19 +171,34 @@ impl ArrowEngineData {
         Ok(ArrowEngineData::new(data?))
     }
 
-    // TODO needs to apply the schema to the parquet read
-    pub fn try_create_from_parquet(_schema: SchemaRef, location: Url) -> DeltaResult<Self> {
+    pub fn try_create_from_parquet(schema: SchemaRef, location: Url) -> DeltaResult<Self> {
         let file = File::open(
             location
                 .to_file_path()
                 .map_err(|_| Error::generic("can only read local files"))?,
         )?;
-        let builder = ParquetRecordBatchReaderBuilder::try_new(file)?;
+        let metadata = ArrowReaderMetadata::load(&file, Default::default())?;
+        let parquet_schema = metadata.schema();
+        let requested_schema: ArrowSchema = (&*schema).try_into()?;
+        let mut builder = ParquetRecordBatchReaderBuilder::try_new(file)?;
+        let indicies: Vec<usize> = get_requested_indicies(&requested_schema, parquet_schema)?;
+        if let Some(mask) = generate_mask(
+            &requested_schema,
+            parquet_schema,
+            builder.parquet_schema(),
```

Comment:
There are two types of schema. 

There's an arrow "schema" (named `parquet_schema` here) which is [this type](https://docs.rs/arrow/latest/arrow/datatypes/struct.Schema.html) in the actual "arrow" crate. And then there's what the parquet crate calls a [SchemaDescriptor](https://arrow.apache.org/rust/parquet/schema/types/struct.SchemaDescriptor.html), which is in the parquet crate and is more low level and tied to the actual parquet format, which is what the `builder` returns from `parquet_schema()`.

I think we could maybe get away with just the `SchemaDescriptor` and reading out names of the columns from the column path descriptors, but it would be more complex.